implementation/contracts/system/TBTCSystem.sol

```diff
@@ -266,6 +268,7 @@ contract TBTCSystem is Ownable, ITBTCSystem, DepositLog {
         payable
         returns (address)
     {
+        require(tbtcDepositToken.exists(uint256(msg.sender)), "Caller must be a Deposit contract");
```

Comment:
It's a nice solution! 👌 I'm glad you took this issue, as I'm way behind the latest changes to deposit tokens.

Reply:
This is a delightfully elegant solution---I know we'd already talked about it, but it's still awesome to see it work so cleanly :)